start_admin_sshd.sh

```diff
@@ -1,30 +1,18 @@
 #!/bin/bash
-# This file is part of Bottlerocket.
-# Copyright Amazon.com, Inc., its affiliates, or other contributors. All Rights Reserved.
-# SPDX-License-Identifier: Apache-2.0 OR MIT
+
 set -e
 
 log() {
   echo "$*" >&2
 }
 
-declare -r local_user="ec2-user"
-declare -r persistent_storage_base_dir="/.bottlerocket/host-containers/current"
-declare -r ssh_host_key_dir="${persistent_storage_base_dir}/etc/ssh"
-declare -r user_data="${persistent_storage_base_dir}/user-data"
-declare -r user_ssh_dir="/home/${local_user}/.ssh"
-available_auth_methods=0
-
-mkdir -p "${user_ssh_dir}"
-chmod 700 "${user_ssh_dir}"
-
 get_user_data_keys() {
     # Extract the keys from user-data json
     local raw_keys
     local key_type="${1:?}"
     # ? signifies an optional object identifier-index and doesn't return a 'failed to iterate' error
     # in the event that a key_type is missing. If jq returns nothing, then that is caught by the -z.
-    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${user_data}") \
+    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${USER_DATA}") \
     || [[ -z "${raw_keys}" ]]; then
```

Comment:
Rather than combining this in the same if, can you split this out into its own branch?

I've added some logging messages to help demonstrate the difference, not necessarily to suggest adding the messages:

```bash
if ! raw_keys=$(jq ...); then
  # can't extract keys from provided data
  log "unable to parse configuration from user-data"
  return 1
fi

if [[ -z "$raw_keys" ]]; then
  # no keys defined
  log "user-data does not provide $raw_keys"
  return 1
fi
```